src/core/ddsi/src/ddsi_udp.c

```diff
@@ -530,6 +641,12 @@ static dds_return_t ddsi_udp_create_conn (struct ddsi_tran_conn **conn_out, stru
   if (gv->config.dontRoute && set_dont_route (gv, sock, ipv6) != DDS_RETCODE_OK)
     goto fail_w_socket;
 
+  // IP_PKTINFO on socket so we get to know the destination address and the interface
+  // on which the packet was received.  If it doesn't work, we don't mind: it simply
+  // means there is slightly less information available for making sense of addresses
+  // or deciding whether multicast SPDP packet really is to be processed
+  (void) setsockopt_pktinfo (gv, sock, ipv6);
```

Comment:
I did a bit of checking using `ddsperf -TOU` on my mac, an ancient windows 10 box and an RPi3 in 64-bit mode with a Linux 5.1. This yields:

* rpi: 74.24 → 71.57  — -3.6%
* mac: 3429.39 → 3350.57 — -2.3%
* windows: 1.34 → 1.36 — measurement noise

(Windows is based on the output of `pub` rather than `sub` and has a different scale.)

I'm tempted to say the benefits outweigh the cost, but I am considering adding an option to disable it altogether.

Reply:
Yes, I agree, an option to disable it would be nice. The default can be 'enabled' because of the limited impact.